MapleServer2/PacketHandlers/Game/ClubHandler.cs

```diff
@@ -104,22 +116,123 @@ private void HandleJoin(GameSession session, PacketReader packet)
             }
         }
 
-        private void HandleInvite(GameSession session, PacketReader packet)
+        private void HandleSendInvite(GameSession session, PacketReader packet)
         {
             long clubId = packet.ReadLong();
             string invitee = packet.ReadUnicodeString();
-            session.Send(ClubPacket.InviteSentReceipt(clubId, invitee));
+
+            Player other = GameServer.Storage.GetPlayerByName(invitee);
+            if (other == null)
+            {
+                return;
+            }
+
+            Club club = GameServer.ClubManager.GetClubById(clubId);
+            if (club == null)
+            {
+                return;
+            }
+
+            session.Send(ClubPacket.InviteSentReceipt(clubId, other.Session.Player));
+            other.Session.Send(ClubPacket.Invite(club, other.Session.Player));
         }
 
         private void HandleInviteResponse(GameSession session, PacketReader packet)
         {
             long clubId = packet.ReadLong();
             string clubName = packet.ReadUnicodeString();
             string clubLeader = packet.ReadUnicodeString();
-            packet.ReadUnicodeString(); //playerName. TODO: verify player name
-            byte response = packet.ReadByte(); // 0 = accept
-            session.Send(ClubPacket.InviteResponse(clubId, clubName, clubLeader, session.Player, response));
-            //TODO handle response
+            string invitee = packet.ReadUnicodeString(); //playerName. TODO: verify player name
+            byte response = packet.ReadByte(); // 1 = accept
+
+            Club club = GameServer.ClubManager.GetClubById(clubId);
+            if (club == null)
+            {
+                return;
+            }
+
+            Player other = GameServer.Storage.GetPlayerByName(invitee);
+            if (other == null)
+            {
+                return;
+            }
+
+            if (response == 1)
+            {
+                club.Leader.Session.Send(ClubPacket.LeaderInviteResponse(clubId, invitee, response));
+                club.BroadcastPacketClub(ClubPacket.ConfirmInvite(clubId, other.Session.Player, response, clubLeader));
+                other.Session.Send(ClubPacket.InviteResponse(clubId, clubName, clubLeader, session.Player, response));
```

Comment:
This should be updated to just take `club` instead of `clubId, clubName, clubLeader`, all of these can just be taken from the `club` itself